### ISSUE TITLE
TypeError: 'type' object is not subscriptable

### ISSUE THREAD

**Issue by nikitaverbis** [adamrehn/ue4-docker#246]
OS: ubuntu 20.04
python ver.: 3.8
ue4-docker: 0.0.97

```
ue4-docker info
/usr/lib/python3/dist-packages/requests/__init__.py:89: RequestsDependencyWarning: urllib3 (1.26.9) or chardet (3.0.4) doesn't match a supported version!
  warnings.warn("urllib3 ({}) or chardet ({}) doesn't match a supported "
Traceback (most recent call last):
  File "/usr/local/bin/ue4-docker", line 5, in <module>
    from ue4docker import main
  File "/usr/local/lib/python3.8/dist-packages/ue4docker/__init__.py", line 1, in <module>
    from .build import build
  File "/usr/local/lib/python3.8/dist-packages/ue4docker/build.py", line 2, in <module>
    from .infrastructure import *
  File "/usr/local/lib/python3.8/dist-packages/ue4docker/infrastructure/__init__.py", line 7, in <module>
    from .ImageBuilder import ImageBuilder
  File "/usr/local/lib/python3.8/dist-packages/ue4docker/infrastructure/ImageBuilder.py", line 11, in <module>
    class ImageBuilder(object):
  File "/usr/local/lib/python3.8/dist-packages/ue4docker/infrastructure/ImageBuilder.py", line 143, in ImageBuilder
    env: Optional[dict[str, str]] = None,
TypeError: 'type' object is not subscriptable

```
Some strange error

**Comment by nikitaverbis** [comment]
Problem can be fixed by changing
```
    from typing import Optional
    ....
    env: Optional[dict[str, str]] = None
```
to 
```
     from typing import Optional, Dict
     ...
     env: Optional[Dict[str, str]] = None
```
in file "ue4docker/infrastructure/ImageBuilder.py", lines 1 and 143.

**Comment by nikitaverbis** [comment]
ue4-docker info (output after fix):
```
/usr/lib/python3/dist-packages/requests/__init__.py:89: RequestsDependencyWarning: urllib3 (1.26.9) or chardet (3.0.4) doesn't match a supported version!
  warnings.warn("urllib3 ({}) or chardet ({}) doesn't match a supported "
ue4-docker version:         0.0.97 (latest available version is 0.0.97)
Operating system:           Linux (Ubuntu 20.04.4 LTS, 5.13.0-40-generic)
Docker daemon version:      20.10.14
NVIDIA Docker supported:    No
Maximum image size:         No limit detected
Available disk space:       1.14 TiB
Total system memory:        31.29 GiB physical, 2 GiB virtual
CPU:                        4 physical, 8 logical (x86_64)
```

**Comment by TBBle** [comment]
Ah yes, using `dict` like that needs [PEP-585](https://peps.python.org/pep-0585/) which was introduced in Python 3.9.

**Comment by slonopotamus** [comment]
Whoops. Fixed and released as ue4-docker 0.0.98.

**Comment by TBBle** [comment]
Interesting. We build and publish packages with Python 3.7, but don't currently run a linter that would catch this. [A trivial search](https://github.com/marketplace?type=actions&query=mypy+) shows a few mypy and multi-linter GitHub Actions, perhaps we could trivially drop one of those into CI? That's assuming they'd actually pick up something like this; `mypy` is just my first guess since it's the tool that I know checks and consumes those annotations, although to be honest I haven't ever run it myself.
